internal/fileutils/fileutils_test.go

```diff
@@ -0,0 +1,156 @@
+package fileutils_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/fileutils"
+)
+
+func TestFileExists(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		name            string
+		fileExists      bool
+		parentDirIsFile bool
+
+		wantExists bool
+		wantError  bool
+	}{
+		"Returns_true_when_file_exists":                      {fileExists: true, wantExists: true},
+		"Returns_false_when_file_does_not_exist":             {fileExists: false, wantExists: false},
+		"Returns_false_when_parent_directory_does_not_exist": {fileExists: false, wantExists: false},
+
+		"Error_when_parent_directory_is_a_file": {parentDirIsFile: true, wantError: true},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			tempDir := t.TempDir()
+			path := filepath.Join(tempDir, "file")
+			if tc.fileExists {
+				err := fileutils.Touch(path)
+				require.NoError(t, err, "Touch should not return an error")
+			}
+			if tc.parentDirIsFile {
+				path = filepath.Join(tempDir, "file", "file")
+				err := fileutils.Touch(filepath.Join(tempDir, "file"))
+				require.NoError(t, err, "Touch should not return an error")
+			}
+
+			exists, err := fileutils.FileExists(path)
+			if tc.wantError {
+				require.Error(t, err, "FileExists should return an error")
+			} else {
+				require.NoError(t, err, "FileExists should not return an error")
+			}
+			require.Equal(t, tc.wantExists, exists, "FileExists should return the expected result")
```

Comment:
We shouldn't need to assert on the `exists` value if the function returned an error, so let's use the early return way instead (?)

Reply:
I think it makes sense to also test that the `exists` value is `false` when an error is returned.

Reply:
This is basically asserting on the zero-value of the variable, so we technically could do that on every test that we have. What I meant is that the "primary" return should not be considered (as it can't be "trusted" since something went wrong in the function, hence the error), and asserting on it might give the wrong impression that it matters, but I won't push for the change.